app/controllers/amr_heatmap_controller.rb

```diff
@@ -0,0 +1,45 @@
+class AmrHeatmapController < ApplicationController
+  include ReportHelper
+
+  before_action :admin_required
+
+  current_power do # Put this here for CLI
+    Power.new(current_user)
+  end
+
+  power :viewable_samples, as: :samples_scope
+
+  # first we grab the Sample from the id parameter passed to us
+  before_action :set_sample
+
+  # GET /amr_heatmap.json
+  # Return JSON information required to create a visualization,
+  # from submitted parameters
+  def index
+    @pipeline_run = select_pipeline_run(@sample, params[:pipeline_version])
```

Comment:
I think it would make more sense to have this function accept multiple sample_ids, as opposed to a single one. If it only accepts a single id, we'd have to call the endpoint multiple times from the front-end in order to get the amr_counts for all of our samples.

Also, I don't think we need to allow for a `pipeline_version` param. We can just use the most recent pipeline_run for each sample, which you can get by calling `sample.first_pipeline_run` (it's poorly named).

Reply:
Done!